### PR TITLE
[VGNWB-370_1] add api to check reset token

### PR DESCRIPTION
**Pull request by sudheerad9**
# Guidelines for Pull Requests

If you haven't yet read our code review guidelines, please do so, You can find them [here](https://diging.atlassian.net/wiki/spaces/DIGING/pages/2256076801/Code+Review+Guidelines).

Please confirm the following by adding an x for each item (turn `[ ]` into `[x]`).

- [x] I have removed all code style changes that are not necessary (e.g. changing blanks across the whole file that don’t need to be changed, adding empty lines in parts other than your own code)
- [x] I am not making any changes to files that don’t have any effect (e.g. imports added that don’t need to be added)
- [x] I do not have any sysout statements in my code or commented out code that isn’t needed anymore
- [x] I am not reformatting any files in the wrong format or without cause. 
- [x] I am not changing file encoding or line endings to something else than UTF-8, LF
- [x] My pull request does not show an insane amount of files being changed although my ticket only requires a few files being changed
- [x] I have added Javadoc/documentation where appropriate
- [x] I have added test cases where appropriate
- [x] I have explained any part of my code/implementation decisions that is not be self-explanatory

## Please provide a brief description of your ticket 
(you can copy the ticket if it hasn't changed) 

(https://diging.atlassian.net/jira/software/c/projects/VGNWB/boards/16?modal=detail&selectedIssue=VGNWB-370)
  
## Anything else the reviewer needs to know?

... describe here ...

### CODE REVIEW

File: accounts/views.py [hunk 1]
```diff
@@ -187,4 +203,4 @@ def mark_as_deleted(self, request, pk=None):
 	def mark_all_as_deleted(self, request):
 		notifications = Notification.objects.filter(recipient=request.user, deleted=False)
 		notifications.mark_all_as_deleted()
-		return Response({})
+		return Response({})
```
Comment (tjquinn1) [inline accounts/views.py]: return a status.

File: accounts/views.py [hunk 2]
```diff
@@ -80,6 +80,22 @@ def citesphere_token(request):
...
+			user = VogonUser.objects.get(username=username)
+		except VogonUser.DoesNotExist:
+			return Response({ "success": False, "message": "User not found" }, status=404)
+	token = request.data.get('token')
+	if token:
+		try:
+			token = ResetToken.objects.get(user=user, token=token)
+		except ResetToken.DoesNotExist:
+			return Response({ "success": False, "message": "Reset token not found" }, status=404)
+	return Response({"success": True, "message": "Token exists for the user"})
```
Comment (tjquinn1) [inline accounts/views.py]: return a status

**Review by tjquinn1 — CHANGES_REQUESTED** [review]

File: accounts/views.py [hunk 3]
```diff
@@ -80,6 +80,22 @@ def citesphere_token(request):
...
+		try:
+			user = VogonUser.objects.get(username=username)
+		except VogonUser.DoesNotExist:
+			return Response({ "success": False, "message": "User not found" }, status=404)
+	token = request.data.get('token')
+	if token:
+		try:
+			token = ResetToken.objects.get(user=user, token=token)
+		except ResetToken.DoesNotExist:
+			return Response({ "success": False, "message": "Reset token not found" }, status=404)
```
Comment (tjquinn1) [inline accounts/views.py]: use django rest status's. status=status.HTTP_404_NOT_FOUND. This way the implementation is always standardized.

File: accounts/views.py [hunk 4]
```diff
@@ -80,6 +80,22 @@ def citesphere_token(request):
...
 		return Response(status=status.HTTP_201_CREATED)
 
+@api_view(['POST'])
+def check_reset_token(request):
+	username = request.data.get('username', None)
+	if username:
+		try:
+			user = VogonUser.objects.get(username=username)
+		except VogonUser.DoesNotExist:
+			return Response({ "success": False, "message": "User not found" }, status=404)
```
Comment (tjquinn1) [inline accounts/views.py]: use django rest status's. status=status.HTTP_404_NOT_FOUND. This way the implementation is always standardized.

**Review by tjquinn1 — CHANGES_REQUESTED** [review]

**Comment by tjquinn1** [comment]
Do you have a script to trigger all the resets?

**Review by jophals — APPROVED** [review]
Approved

**Pull request merged**